app/mailers/user_mailer.rb

```diff
@@ -159,8 +164,11 @@ def account_reset_request(account_reset)
 
   def account_reset_granted(account_reset)
     with_user_locale(user) do
+      presenter = ConfirmationEmailPresenter.new(user, view_context)
+      @confirmation_period = presenter.confirmation_period
       @token = account_reset&.request_token
       @granted_token = account_reset&.granted_token
+      @account_reset_deletion_period_hours = account_reset_deletion_period_hours
```

Comment:
Similar question as prior comment: The `ConfirmationEmailPresenter` uses the timeframe for _account confirmation_, but this is related to account reset, so we'd want to use `account_reset_wait_period_days` ?

Reply:
I need both `confirmation_period` and `account_reset_deletion_period_hours`. 

Account reset days is shown in "Your 24 hour waiting period has ended, please complete step 2 of the process." `confirmation_period` is then needed for "This link will expire in 24 hours". I understood that link expiration will always be at 24 hours and it should stay that way.

Reply:
If what I have is the case, then it would be clearer to change `confirmation_period` to something like `link_expiration_period`

Reply:
> I need both `confirmation_period` and `account_reset_deletion_period_hours`.
> 
> Account reset days is shown in "Your 24 hour waiting period has ended, please complete step 2 of the process." `confirmation_period` is then needed for "This link will expire in 24 hours". I understood that link expiration will always be at 24 hours and it should stay that way.

Good call on the difference between delay + link valid period.

But I think the link validity period is based on another configuration, not `Devise.confirm_within`, at least based on what I'm seeing here which is part of the validation for the link token:

https://github.com/18F/identity-idp/blob/dbc3c15bc07f8fe1dfe5d82407edb7cb81bef3d3/app/models/account_reset_request.rb#L16-L20

Reply:
It took me a minute, but I see it now. I can make the change.